assets/js/base/components/payment-methods/payment-methods.js

```diff
@@ -64,38 +62,36 @@ const getPaymentMethod = ( id, paymentMethods, isEditor ) => {
 };
 
 const PaymentMethods = () => {
-	const [ checkoutData ] = useCheckoutData();
 	const { isEditor } = useCheckoutContext();
-	const { dispatch, select } = usePaymentEvents();
 	const { isInitialized, paymentMethods } = usePaymentMethods();
 	const currentPaymentMethods = useRef( paymentMethods );
+	const {
+		activePaymentMethod,
+		setActivePaymentMethod,
+		...paymentMethodInterface
+	} = usePaymentMethodInterface();
+	const currentPaymentMethodInterface = useRef( paymentMethodInterface );
 
 	// update ref on changes
 	useEffect( () => {
 		currentPaymentMethods.current = paymentMethods;
-	}, [ paymentMethods ] );
-
-	const {
-		activePaymentMethod,
-		setActivePaymentMethod,
-	} = useActivePaymentMethod();
+		currentPaymentMethodInterface.current = paymentMethodInterface;
+	}, [ paymentMethods, paymentMethodInterface, activePaymentMethod ] );
 	const getRenderedTab = useCallback(
 		() => ( selectedTab ) => {
 			const paymentMethod = getPaymentMethod(
 				selectedTab,
 				currentPaymentMethods.current,
 				isEditor
 			);
-			const paymentEvents = { dispatch, select };
 			return paymentMethod
 				? cloneElement( paymentMethod, {
-						isActive: true,
-						checkoutData,
-						paymentEvents,
+						activePaymentMethod: paymentMethod.id,
```

Comment:
I've seen that payment methods sometimes have a `slug`, other times an `id` (and these is used as `key` and `name` in components). If we can standardise on one name for this id / slug (slug?) that might help simplify this.

Reply:
Good call on this. I looked for places where this was inconsistent and addressed (see ff5de1)